src/container/getRelayQueries.js

```diff
@@ -46,9 +46,16 @@ function getRelayQueries(
     return cache[cacheKey];
   }
   var querySet = {};
-  Component.getFragmentNames().forEach(fragmentName => {
-    // TODO: Fix this. It relies on the query and fragment names matching.
-    var queryName = fragmentName;
+  Object.keys(route.queries).forEach(queryName => {
+    var fragmentName = queryName;
+    invariant(
+      Component.hasFragment(fragmentName),
+      'getRelayQueries: Route `%s` expected `%s` to have a fragment named ' +
+      '`%s`.',
+      route.name,
+      Component.displayName,
+      fragmentName
+    );
```

Comment:
GH won't let me add a comment to a non-updated line, so can you also change line 68-69 to:

```javascript
invariant(
  ...,
  'Relay.QL: query `%s.queries.%s` is invalid, a typical query is defined using: ' +
  '() => Relay.QL`query { ... }`.',
  route.name,
  queryName
```